OpenRA.Mods.Common/Traits/Armament.cs

```diff
@@ -62,6 +62,19 @@ public class ArmamentInfo : UpgradableTraitInfo, Requires<AttackBaseInfo>
 		[Desc("Use multiple muzzle images if non-zero")]
 		public readonly int MuzzleSplitFacings = 0;
 
+		[Desc("Is it possible to use the armament when force-attacking?")]
+		public readonly bool ForceAttacks = true;
+
+		[Desc("Is the armament beneficial (should it be used on allies)?")]
+		public readonly bool Beneficial = false;
```

Comment:
A suggestion:

"NotHarmful" instead of "Beneficial" ?
That would imply that a unit wouldn't care about getting hit by it, regardless of its diplomatic stance.

A healing serum would be not harmful, while a targeting laser would be harmful (even though it doesn't directly cause harm, it still results in harm.)